connectors/sources/onedrive.py

```diff
@@ -113,34 +166,100 @@ def set_logger(self, logger_):
         self._logger = logger_
 
     @cached_property
-    def _get_session(self):
+    def session(self):
         """Generate base client session with configuration fields
         Returns:
             ClientSession: Base client session
         """
-
-        timeout = aiohttp.ClientTimeout(total=300)
+        connector = aiohttp.TCPConnector(limit=DEFAULT_PARALLEL_CONNECTION_COUNT)
+        timeout = aiohttp.ClientTimeout(total=REQUEST_TIMEOUT)
         return aiohttp.ClientSession(
-            timeout=timeout,
-            raise_for_status=True,
+            timeout=timeout, raise_for_status=True, connector=connector
         )
 
     async def close_session(self):
         self._sleeps.cancel()
-        await self._get_session.close()
-        del self._get_session
+        await self.session.close()
+        del self.session
 
     @retryable(
         retries=RETRIES,
         interval=RETRY_INTERVAL,
         strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
     )
-    async def get(self, url_name):
-        url = parse.urljoin(BASE_URL, url_name)
+    async def get(self, url):
         access_token = await self.token.get()
         headers = {"authorization": f"Bearer {access_token}"}
-        async with self._get_session.get(url=url, headers=headers) as response:
-            yield response
+        try:
+            async with self.session.get(url=url, headers=headers) as response:
+                yield response
+        except ServerConnectionError:
+            await self.close_session()
```

Comment:
I think it won't retry because it doesn't raise anything.

Reply:
You're right. I missed it 🤦‍♂️, fixed that now.